docs/recipes/recipes.md

```diff
@@ -889,6 +889,81 @@ Example := Window {
 }
 ```
 
+### Breakpoints for responsive ui
+
+Uses this recipe to implement a responsive SideBar that collapses when the parent width is smaller then the given break-point. On Button
+click the SideBar can be expanded if collapsed on break-point.
+
+```slint,no-auto-preview
+import { Button } from "std-widgets.slint";
+
+export SideBar := Rectangle {
+    private property <bool> collapsed: reference-width < break-point;
+    property <length> reference-width;
+    property <length> break-point: 600px;
```

Comment:
Since this is copy & paste ready, I wonder if it should have basically "API" documentation via comments here, and comments down at the use-site how it is used.

Perhaps the text of the button should be a configurable property?

That said, I guess this needs editing anyway as users may want to add more than just one button (like you did in your clip).

Reply:
It's a good suggestion. I will improve it. Thanks.